### ISSUE TITLE
StepTypeError when different step types have the same step name

### ISSUE THREAD

**Issue by lrowe** [pytest-dev/pytest-bdd#122]
When a step is registered for both when and then it throws a StepTypeError because the first step registration decorator (first decorator runs last) overwrites the step registration of the other step registration. Similar to #120 

Test case: https://github.com/lrowe/pytest-bdd/tree/same-step-name

Interestingly the builtin ``trace`` decorator which is registered for all of given/when/then does appear to work for ``When trace`` and ``Then trace``: https://github.com/pytest-dev/pytest-bdd/blob/2.10.0/pytest_bdd/plugin.py#L24

That seems to be due to the function name being the same as the step name and the ``Given trace``  fixture being incorrectly loaded. Taking an example:

```cucumber
Feature: Trace
    Scenario: When trace
        When trace
```
```python
from pytest_bdd import scenario


@scenario('trace.feature', 'When trace')
def test_when_trace():
    pass
```

Continuing from ``When trace`` throws an AttributeError when I would expect the test to pass:
```
$ .tox/py34/bin/py.test test_trace.py --tb native
========================================== test session starts ===========================================
platform darwin -- Python 3.4.2 -- py-1.4.28 -- pytest-2.7.1 -- /Users/lrowe/scratch/pytest-bdd/.tox/py34/bin/python3.4
cachedir: /Users/lrowe/scratch/pytest-bdd/.cache
rootdir: /Users/lrowe/scratch/pytest-bdd, inifile: tox.ini
plugins: bdd, cache, cov, pep8, xdist
collected 1 items 

test_trace.py::test_when_trace 
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
--Return--
> /Users/lrowe/scratch/pytest-bdd/pytest_bdd/plugin.py(29)trace()->None
-> pytest.set_trace()
(Pdb) c
FAILED

================================================ FAILURES ================================================
____________________________________________ test_when_trace _____________________________________________
Traceback (most recent call last):
  File "/Users/lrowe/scratch/pytest-bdd/test_trace.py", line 5, in test_when_trace
    def test_when_trace():
  File "/Users/lrowe/scratch/pytest-bdd/pytest_bdd/scenario.py", line 179, in _execute_scenario
    if step_func.step_type != step.type:
AttributeError: 'NoneType' object has no attribute 'step_type'
======================================== 1 failed in 1.24 seconds ========================================
```

Renaming the function ``def trace():`` in plugin.py to ``def trace2():`` exposes the underlying error reported here:
```
$ .tox/py34/bin/py.test test_trace.py --tb native
========================================== test session starts ===========================================
platform darwin -- Python 3.4.2 -- py-1.4.28 -- pytest-2.7.1 -- /Users/lrowe/scratch/pytest-bdd/.tox/py34/bin/python3.4
cachedir: /Users/lrowe/scratch/pytest-bdd/.cache
rootdir: /Users/lrowe/scratch/pytest-bdd, inifile: tox.ini
plugins: bdd, cache, cov, pep8, xdist
collected 1 items 

test_trace.py::test_when_trace FAILED

================================================ FAILURES ================================================
____________________________________________ test_when_trace _____________________________________________
Traceback (most recent call last):
  File "/Users/lrowe/scratch/pytest-bdd/test_trace.py", line 5, in test_when_trace
    def test_when_trace():
  File "/Users/lrowe/scratch/pytest-bdd/pytest_bdd/scenario.py", line 181, in _execute_scenario
    'Wrong step type "{0}" while "{1}" is expected.'.format(step_func.step_type, step.type)
pytest_bdd.exceptions.StepTypeError: Wrong step type "given" while "when" is expected.
======================================== 1 failed in 0.01 seconds ========================================
```

**Comment by bubenkoff** [comment]
right
im looking into it now

**Comment by lrowe** [comment]
Another approach to this might be to have a ``@step`` like behave which registers for all of Given/When/Then. That might be more complicated for pytest-bdd as Given registers differently.

(Cucumber just treats all of Given/When/Then as interchangeable, though I prefer behave's approach of allowing universal registration with '@step'.)

**Comment by bubenkoff** [comment]
solved with #121

**Comment by bubenkoff** [comment]
2.11.0 is out
